source/init/mus_initFluid_module.f90

```diff
@@ -277,17 +217,136 @@ subroutine mus_init_advRel_fluid( relaxation, layout, compute )
       case ('d2q9')
         compute => bgk_Regularized_d2q9
       case default
-        write(logUnit(1),*) 'Stencil '//trim(layout)//' is not supported yet!'
-        call tem_abort()
+        call tem_abort('Stencil '//trim(layout)//' is not supported yet!')
       end select
 
     case default
-      write(logUnit(1),*) 'Relaxation '//trim(relaxation)//' is not supported!'
-      call tem_abort()
+      call tem_abort('Relaxation '//trim(relaxation)//' is not supported!')
     end select
 
   end subroutine mus_init_advRel_fluid
   ! **************************************************************************** !
 
+  ! ************************************************************************** !
+  !> This routine assigns compute routine for bgk relaxation
+  subroutine mus_init_advRel_fluid_bgk( variant, layout, compute )
+    ! --------------------------------------------------------------------------
+    character(len=labelLen), intent(in) :: variant
+    character(len=labelLen), intent(in) :: layout
+    procedure( kernel ), pointer, intent(out) :: compute
+    ! --------------------------------------------------------------------------
+    select case(trim(variant))
+    case ('standard')
+      select case (trim(layout))
+      case ('d3q27')
+        compute => mus_advRel_kCFD_rBGK_vStd_lD3Q27
+      case ('d3q19')
+        compute => mus_advRel_kFluid_rBGK_vStd_lD3Q19
+      case ('d2q9')
+        compute => mus_advRel_kFluid_rBGK_vStd_lD2Q9
+      case default
+        call tem_abort('Unsupported variant "'//trim(variant)//'" for'// &
+          &             ' "bgk" relaxation!')
+      end select
+
+    case ('standard_no_opt')
+      compute => mus_advRel_kCFD_rBGK_vStdNoOpt_l
+
+    case ('improved')
+      select case (trim(layout))
+      case ('d3q27')
+        compute => mus_advRel_kFluid_rBGK_vImproved_lD3Q27
+      case ('d2q9')
+        compute => mus_advRel_kFluid_rBGK_vImproved_lD2Q9
+      case default
+        call tem_abort('Stencil '//trim(layout)//' is not supported yet!')
+      end select
+
+    case ('block')
+      select case (trim(layout))
+      case ('d3q19')
+        compute => mus_advRel_kFluid_rBGK_vBlock_lD3Q19
+      case default
+        call tem_abort('Stencil '//trim(layout)//' is not supported yet!')
```

Comment:
I'd prefer to refer to the name we use in the UI, which to my understanding is `layout`, instead of `stencil`.